Simplenote/NoteEditorViewController.m

```diff
@@ -293,8 +293,11 @@ - (void)displayNotes:(NSArray *)notes
 - (void)checkTextInDocument
 {
     dispatch_async(dispatch_get_main_queue(), ^() {
+        // Disable editor delegate while processing links to prevent the editor from modifying the note
```

Comment:
can we add a little more explanation how `checkTextInDocument` modifies the note?

Reply:
I updated it @dmsnell, does it make more sense now?

Reply:
definitely better, but I'm still not sure why that changes the note - does it only change it if the editor has unsaved changes? I'm probably just ignorant - don't let that hold up this PR

Reply:
No problem. `checkTextInDocument` is a poorly named native API that does formatting changes to the text in the editor, so it makes links clickable and runs the spell checker. It's firing a textDidChange notification, which is what we use to save a note after a user makes a change to it (like types in a new character).

https://github.com/Automattic/simplenote-macos/blob/7680fc1323143581994902785d9a5ae0378fadb4/Simplenote/NoteEditorViewController.m#L449

Reply:
so this would mean that every character entered in would sync a new revision in Simperium? is that the error? or is the error the link attributes in the note that iOS adds?

Reply:
The content of the note doesn't change when this fires, but the modification date does. (and this is for macOS)

Reply:
> so this would mean that every character entered in would sync a new revision in Simperium?

No, we use the same two second save timer delay like in electron. All the apps do that.

Reply:
ah, so we open a note, then `checkTextInDocument` fires, then the modification date changes and the note jumps to the top of the list even though all we did was read it? that makes sense

Reply:
Yes, that's the issue. And this wasn't caught because the macOS app doesn't move a note up in the list even after it is modified so that it keeps its place in the notes list. I wonder if we should change that behavior.